impeller/BUILD.gn

```diff
@@ -80,6 +79,10 @@ impeller_component("impeller_unittests") {
     ]
   }
 
+  if (impeller_enable_metal) {
+    deps += [ "//flutter/impeller/renderer/backend/metal:metal_unittests" ]
+  }
```

Comment:
No need to build these guys on windows, linux, or custom embedders for platforms without metal enabled.